httpbakery/client.go

```diff
@@ -125,6 +125,28 @@ func DischargeAll(m *macaroon.Macaroon, client *http.Client, visitWebPage func(u
 	return bakery.DischargeAll(m, ctxt.obtainThirdPartyDischarge)
 }
 
+// GetPublicKey returns the public key provided by a server which is using
+// httpbakery.AddDischargeHandler.
+func GetPublicKey(url string) (*bakery.PublicKey, error) {
+	resp, err := http.Get(url + "/publickey")
+	if err != nil {
+		return nil, errgo.Notef(err, "failed to make request for public key")
+	}
+	if resp.StatusCode != http.StatusOK {
+		return nil, errgo.Newf("server returned error code %v", resp.StatusCode)
+	}
+	defer resp.Body.Close()
+	decoder := json.NewDecoder(resp.Body)
```

Comment:
In general for JSON response bodies, I prefer to do an ioutil.ReadAll - that way we don't need to abandon the network connection if the JSON happens to have a trailing newline.